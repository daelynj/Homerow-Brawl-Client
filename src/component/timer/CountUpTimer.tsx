import * as React from "react";
import { useState, useEffect } from "react";

interface Props {
  setFinishTime: (newTime: number) => void;
}

export const CountUpTimer = (props: Props) => {
  const [seconds, setSeconds] = useState<number>(0);

  useEffect(() => {
    let timerID = setTimeout(tick, 1000);

    return function cleanup() {
      props.setFinishTime(seconds);
    };
  });

  const tick = () => {
    setSeconds(seconds + 1);
  };

  return (
    <>
      <div>Go!</div>
      Timer: {seconds}
    </>
  );
};
