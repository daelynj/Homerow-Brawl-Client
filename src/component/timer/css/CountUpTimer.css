.countup {
  font-size: 20pt;
  margin-left: auto;
  margin-right: auto;
  width: 30%;
  border: 5px solid rgb(98, 68, 57);
  text-align: center;
}
