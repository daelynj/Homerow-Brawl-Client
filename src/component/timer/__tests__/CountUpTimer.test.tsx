import * as React from "react";
import { CountUpTimer } from "../CountUpTimer";
import renderer from "react-test-renderer";

describe("CountUpTimer", () => {
  jest.useFakeTimers();
  const setFinishTime = jest.fn();
  const setMakeRequests = jest.fn();

  it("renders the CountUpTimer", () => {
    const tree = renderer
      .create(
        <CountUpTimer
          setMakeRequests={setMakeRequests}
          setFinishTime={setFinishTime}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls tick every 1000 miliseconds", () => {
    const TimerComponent = renderer.create(
      <CountUpTimer
        setMakeRequests={setMakeRequests}
        setFinishTime={setFinishTime}
      />
    );

    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);

    TimerComponent.update(
      <CountUpTimer
        setMakeRequests={setMakeRequests}
        setFinishTime={setFinishTime}
      />
    );

    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);
  });

  it("sets the finish time when unmounted", () => {
    const TimerComponent = renderer.create(
      <CountUpTimer
        setMakeRequests={setMakeRequests}
        setFinishTime={setFinishTime}
      />
    );
    TimerComponent.unmount();

    expect(setFinishTime).toHaveBeenCalled();
  });
});
