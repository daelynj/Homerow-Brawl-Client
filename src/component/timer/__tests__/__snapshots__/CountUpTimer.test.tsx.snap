// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountUpTimer renders the CountUpTimer 1`] = `
<div
  className="countup"
>
  <div>
    Go!
  </div>
  Timer: 
  0
</div>
`;
